src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSTreeNode.java

```diff
@@ -120,15 +120,15 @@ public boolean equals(final Object o) {
         if (length != that.length) return false;
         if (name != null ? !name.equals(that.name) : that.name != null) return false;
         if (rank != null ? !rank.equals(that.rank) : that.rank != null) return false;
-        if (parent != null ? !parent.equals(that.parent) : that.parent != null) return false;
+        if (parent != that.parent) return false;
         return children.equals(that.children);
     }
 
     @Override
     public int hashCode() {
         int result = name != null ? name.hashCode() : 0;
```

Comment:
I think you can use ```Objects.hashCode``` (mind the s in Objects) to handle the null // probrably return 0 for a null anway.

Reply:
So this code could be 
```
return (((Objects.hashCode(name) * 31) + Objects.hashCode(rank) * 31) + parent * 31 ...)
```
Not necesarely in a single line.

Reply:
Actually there is a convenient ```Objects.hash(Object ... o)``` to do the 31 recursion but perhaps  is overkill considering that boxing and Object[] instantiation will take place.

Reply:
This may be faster, but objects with equal hashcodes aren't guaranteed to be equal right?